shared/fs/top-bar/sort.tsx

```diff
@@ -1,17 +1,16 @@
-// @flow
 import * as React from 'react'
 import * as Styles from '../../styles'
 import * as Kb from '../../common-adapters'
 import * as Types from '../../constants/types/fs'
 import * as Flow from '../../util/flow'
 
-export type SortBarProps = {|
-  sortByNameAsc?: ?() => void,
-  sortByNameDesc?: ?() => void,
-  sortByTimeAsc?: ?() => void,
-  sortByTimeDesc?: ?() => void,
-  sortSetting?: ?Types.SortSetting,
-|}
+export type SortBarProps = {
+  sortByNameAsc?: () => void | null
```

Comment:
Does precedence work as expected here? I have a suspicion that TS reads this as `() => (void | null)`

Reply:
good catch!